mlir/include/mlir/Dialect/Bufferization/IR/BufferizableOpInterface.h

```diff
@@ -351,6 +349,11 @@ struct BufferizationOptions {
   /// used.
   UnknownTypeConverterFn unknownTypeConverterFn = nullptr;
 
+  // Use during type conversion to determine the memory space for memref based
+  // on the originanl tensor type if the memory space cannot be inferred.
+  DefaultMemorySpaceFn defaultMemorySpaceFn =
```

Comment:
Add something like this to the docs: `In case of std::nullopt, bufferization fails when the memory space cannot be inferred at any point.`

Reply:
comment updated